ondemand/README.md

```diff
@@ -1,4 +1,389 @@
+
+### Updating the Jupyter App
+
+#### Change bc_queue to a select field
+
+We have 2 partitions enabled in the SLURM containers (SLURM calls queues partitions, so we'll be switching
+back and forth between the two terms in this tutorial). We've started with a field `bc_queue` that
+is a text field, but it's likely much easier for users to simply choose the partition out of a
+select dropdown menu instead.
+
+So let's remove the `bc_queue` field in the form and replace it with a new field we'll call `custom_queue`.
+Add `custom_queue` in the form section and remove bc_queue.  
+
+We'll also add `custom_queue` to the attributes section.  Adding a field to the form section adds it to the form
+in the UI.  By default, this field will be a text field. If you want this field to be a different type of widget
+(as we do) you'll configure the field in the attributes section.  Also by default the label in the UI is the
+same just the name of the field. In our case `custom_queue` would turn into "Custom Queue".  This is only
+slightly correct, so we want to specify the label as "Partition" because that's what it is in SLURM parlance.
+
+Here you can see that we specify the `custom_queue` in the attributes as a select widget with two options.
+and a new label. The first element in options arrays is what will be shown to the user (the capitalized version)
+where the second element is the value what's actually used in the sbatch command.
+
+```yaml
+# other things in form.yml removed for brevity, but should still exist in yours
+attributes:
+  custom_queue:
+    widget: "select"
+    label: "Partition"
+    options:
+      - ["Compute", "compute"]
+      - ["Debug", "debug"]
+form:
+  - custom_queue
+#   - bc_queue
+```
+
+Refresh the [new session form](https://localhost:3443/pun/sys/dashboard/batch_connect/dev/jupyter/session_contexts/new)
+and you should now see your updates.
+
+But before we submit to test them out, we'll need to reconfigure the `submit.yml.erb` to use this
+new field.  You can edit it in the
+[file editor app](https://localhost:3443/pun/sys/file-editor/edit/home/hpcadmin/ondemand/dev/jupyter/submit.yml.erb).
+
+You'll need to specify the script's queue_name as the partition like so. The `script` is the logical
+"script" we're submitting to the scheduler.  And the `queue_name` is the field of the script that will
+specify the queue. (in SLURM parlance it's called partition, so that's why we're calling it partition, but
+OnDemand does this translation at lower layer).
+
+```yaml
+script:
+  queue_name: "<%= custom_queue %>"
+```
+
+Here you'll notice that the file extension is `.erb` and the `<%= custom_queue %>` is a little strange.
+That's because this file is a embedded ruby template file and `<%= custom_queue %>` is ERB template syntax for
+turning the `custom_queue` variable (defined in the `form.yml` and specified by the user when they press "Launch")
+into a string for the yml file.
```

Comment:
I would reduce this to:

The `.erb` file extension indicates this is embedded ruby file. `<%=` and `%>` are embedded ruby tags. A variable for each attribute defined in the form.yml can be used in this ERB file, containing the string representation of the value submitted by the user.

Reply:
Done. Replaced outright with what you've given.